### PR TITLE
Split GraphQL type-def generation.

### PR DESCRIPTION
**Pull request by peterp**
This splits the api and web graphql type def generation into two functions, so that if web fails api can continue, and vice versa.

This also changes the configuration to use `useEnumAsTypes` which provides a useable experience, which should close: #2788

image

### CODE REVIEW

File: packages/internal/src/generate/typeDefinitions.ts [hunk 1]
```diff
@@ -161,57 +165,81 @@ export const generateTypeDefGlobalContext = () => {
...
-        schema: rwjsPaths.generated.schema,
-        config: {
-          scalars: {
-            DateTime: 'string',
-            Date: 'string',
-            JSON: 'Record<string, unknown>',
-            JSONObject: 'Record<string, unknown>',
-            Time: 'string',
+    const rwjsPaths = getPaths()
+    const f = await generateTypeDefGraphQL({
```
Comment (dac09) [inline packages/internal/src/generate/typeDefinitions.ts:171]: Hey peter, totally stylistic, but what do you think of naming these vars a little more explicitly? Sometimes its hard to figure out when you've not looked at the code in a while.

File: packages/internal/src/generate/typeDefinitions.ts [hunk 2]
```diff
@@ -161,57 +165,81 @@ export const generateTypeDefGlobalContext = () => {
...
+}
 
+export const generateTypeDefGraphQLWeb = async () => {
+  if (findCells().length) {
+    const rwjsPaths = getPaths()
+    try {
+      const f = await generateTypeDefGraphQL({
+        [path.join(rwjsPaths.web.base, 'types/graphql.d.ts')]: {
+          documents: './web/src/**/!(*.d).{ts,tsx,js,jsx}',
+          plugins: [
```
Comment (dac09) [inline packages/internal/src/generate/typeDefinitions.ts:197]: I think this is missing the `omitOperationSuffix: true,`, the api one probably doens't need it, but the web one does (not 100% sure)

File: packages/internal/src/generate/typeDefinitions.ts [hunk 3]
```diff
@@ -161,57 +165,81 @@ export const generateTypeDefGlobalContext = () => {
...
+      console.error('Error: Could not generate GraphQL type definitions (web)')
+      console.error()
+      return []
+    }
+  } else {
     return []
   }
 }
+
+const generateTypeDefGraphQL = async (generates: Record<string, unknown>) => {
```
Comment (dac09) [inline packages/internal/src/generate/typeDefinitions.ts:219]: Do we still need this function?

Comment (peterp) [reply packages/internal/src/generate/typeDefinitions.ts:219]: Yup, this contains the majority of the configuration. The other two functions pass their own small samples of configuration in here.

File: packages/internal/src/generate/typeDefinitions.ts (hunk 2 above)
Comment (peterp) [reply packages/internal/src/generate/typeDefinitions.ts:197]: It's lower down over here: https://github.com/redwoodjs/redwood/pull/2839/files#diff-e60ed2b5527fb9736c6f69a43c98c5e71372805eb6bcbe0ee01372854e81086cR235

**Pull request merged**